File: OdontoFlowAPI/OdontFlow/OdontFlow.Domain/ViewModel/User/UserViewModel.cs
namespace OdontFlow.Domain.ViewModel.User;

public class UserViewModel
{
    public Guid Id { get; set; }
    public string Email { get; set; } = default!;
    public string RoleName { get; set;  }
    public int RoleId { get; set; }  
    public bool ChangePassword { get; set; }
    public string Password { get; set; }
    public Guid? EmployeeId { get; set; }
    public string? EmployeeName { get; set; } // ejemplo si quieres mostrar el nombre del empleado
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public bool Active { get; set; }
    public string PasswordHash { get; set; }
    public Guid? ClientId { get; set; }
    public string? ClientName { get; set; }
}

File: OdontoFlowWeb/OdontFlow/src/app/core/model/user-view.model.ts
export interface UserViewModel {
    id?: string;
    email?: string;
    roleId?: number; 
    roleName?: string; 
    changePassword?: boolean;
    employeeId?: string | null;
    employeeName?: string | null;
    clientId?: string | null;
    clientName?: string | null;
    createdAt?: string;
    updatedAt?: string;
    active?: boolean
  }

  
  export interface CreateUserDTO {
    email: string; 
    role?: number;
    changePassword?: boolean;   
    employeeId?: string | null;
    clientId?: string | null;
    active? : boolean
  }

  export interface UpdateUserDTO {
    id: any;
    email?: string;
    role?: number; 
    changePassword?: boolean;
    employeeId?: string | null;
    clientId?: string | null;
    active? : boolean
  }
  
  